test/retcodes_test.py

```diff
@@ -170,3 +170,35 @@ def new_func(*args, **kwargs):
         with mock.patch('luigi.scheduler.Scheduler.add_task', new_func):
             self.run_and_expect('RequiringTask', 0)
             self.run_and_expect('RequiringTask --retcode-not-run 5', 5)
+
+    @with_config({'worker': {'keep_alive': 'True'}})
+    def test_retry_sucess_task(self):
+        class Bar(luigi.Task):
+            num = luigi.IntParameter()
+            has_run = False
```

Comment:
Things become tricky now! Because you'll have more than one instance of this task, and `has_run = False` will be set on the class not the instance. If you just change everything to a parameter-less task like the one above. You don't need to think about these complications.